pandas/core/arrays/period.py

```diff
@@ -1089,3 +1111,38 @@ def _make_field_arrays(*fields):
     ]
 
     return arrays
+
+
+if _PYARROW_INSTALLED and (
+    LooseVersion(pyarrow.__version__) >= LooseVersion("0.14.1.dev")
+):
+
+    class PeriodType(pyarrow.ExtensionType):
+        def __init__(self, freq):
+            # attributes need to be set first before calling
+            # super init (as that calls serialize)
+            self._freq = freq
+            pyarrow.ExtensionType.__init__(self, pyarrow.int64(), "pandas.period")
+
+        @property
+        def freq(self):
+            return self._freq
+
+        def __arrow_ext_serialize__(self):
+            metadata = {"freq": self.freq}
+            return json.dumps(metadata).encode()
+
+        @classmethod
+        def __arrow_ext_deserialize__(cls, storage_type, serialized):
+            metadata = json.loads(serialized.decode())
+            return PeriodType(metadata["freq"])
+
+        def __eq__(self, other):
```

Comment:
Do you also need to implement `__hash__`?

Reply:
Hmm, not sure why it is unhashable right now, as the base DataType class implements a `hash`. As a subclass, if you override `__eq__`, you also need to override `__hash__` ?

Reply:
Yes, that is the case, will add.

Reply:
Yeah, I think that's a Python 3 thing.